apps/manager_app/manager_app.c

```diff
@@ -92,8 +94,38 @@
 /*****************************************************************************
  * GLOBAL FUNCTIONS
  *****************************************************************************/
-void manager_app_main(void) {
-  // This is a dummy implementation, which needs to be updated
-  while (1)
-    ;
-}
\ No newline at end of file
+void manager_app_main(usb_event_t usb_evt) {
+  manager_query_t query = MANAGER_QUERY_INIT_ZERO;
+
+  if (false == decode_manager_query(usb_evt.p_msg, usb_evt.msg_size, &query)) {
+    return;
+  }
+
+  // TODO: Add calls to flows/ functions based on query type decoded from the
+  // protobuf
+  switch ((uint8_t)query.which_request) {
```

Comment:
why typecast explicitly?

Reply:
Type of `which_request` is `pb_size_t` which is incompatible with the switch case statement.

Switch case only allows `uint_x` data types, so explicit typecast is necessary